homeassistant/components/binary_sensor/august.py

```diff
@@ -60,24 +68,55 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     for doorbell in data.doorbells:
         for sensor_type in SENSOR_TYPES:
+            if SENSOR_TYPES[sensor_type][1] == 'door':
+                continue
+            _LOGGER.debug("Adding doorbell sensor class %s for %s",
+                          SENSOR_TYPES[sensor_type][1], doorbell.device_name)
             devices.append(AugustBinarySensor(data, sensor_type, doorbell))
 
+    from august.lock import LockDoorStatus
+    for lock in data.locks:
+        for sensor_type in SENSOR_TYPES:
+            if SENSOR_TYPES[sensor_type][1] != 'door':
+                continue
+
+            state_provider = SENSOR_TYPES[sensor_type][2]
+            if state_provider(data, lock) is LockDoorStatus.UNKNOWN:
```

Comment:
We shouldn't filter out anything, that means that they might just be unavailable during boot and will then require a restart once fixed up by the user. Instead we should return `False` from the available property when status is unknown and still add them to HA.

Reply:
I don't understand how you can have handling in is_on for unknown which is for real devices, and then filter out unknowns that are not real devices. How can you distinguish which is which?

Can't we look at the model and have a flag if they enable sensors?

Reply:
@balloob, 

There does not seem to be a way to really determine if a sensor is there. We can get the model number but that does not automatically mean there is a doorsense sensor as that is added separately as part of the installation for this lock.
Further, if we were to look at the actual model number then we would need to keep on updating based on new models they might provide.